### ISSUE TITLE
mosquitto broker >v1.5.8

### ISSUE THREAD

**Issue by emes** [christianTF/LoxBerry-Plugin-MQTT-Gateway#23]
current version of mosquitto is 1.6.2
gateway fails with any mosquitto broker version after 1.5.8

```
New connection from 192.168.32.101 on port 1883.
New client connected from 192.168.32.101 as Net::MQTT::Simple[2780] (p1, c1, k60, u'loxberry').
Will message specified (12 bytes) (r1, q0).
        loxberry/mqttgateway/status
Sending CONNACK to Net::MQTT::Simple[2780] (0, 0)
Received SUBSCRIBE from Net::MQTT::Simple[2780]
Socket error on client Net::MQTT::Simple[2780], disconnecting.
Sending PUBLISH to mosqsub/2733-loxberry (d0, q0, r0, m0, 'loxberry/mqttgateway/status', ... (12 bytes))
```

And same configuration works fine with mosquitto broker v1.5.8

```
New connection from 192.168.32.101 on port 1883.
New client connected from 192.168.32.101 as Net::MQTT::Simple[949] (c1, k60, u'loxberry').
Will message specified (12 bytes) (r1, q0).
        loxberry/mqttgateway/status
Sending CONNACK to Net::MQTT::Simple[949] (0, 0)
Received SUBSCRIBE from Net::MQTT::Simple[949]
        readaloud/+/+/heartbeat (QoS 0)
        loxberry/mqttgateway/# (QoS 0)
        shellies/+/relay/+ (QoS 0)
Sending SUBACK to Net::MQTT::Simple[949]
```

Doing the same subscribes on the loxberry using mosquitto_sub (v1.4.10) works fine, so not sure if a Net::MQTT::Simple issue or not.

```
loxberry@loxberry:~ $ mosquitto_sub -h mqtt -p 1883 -u loxberry -P xxx -v -t 'shellies/+/relay/+' -t 'loxberry/mqttgateway/#' -t 'readaloud/+/+/heartbeat'
shellies/shelly1-554C0E/relay/0 off
shellies/shelly1-55E297/relay/0 off
loxberry/mqttgateway/keepaliveepoch 1559284861
shellies/shelly1-55E297/relay/0 off
loxberry/mqttgateway/keepaliveepoch 1559284922
shellies/shelly1-554C0E/relay/0 off
readaloud/api/camera-camn/heartbeat 1559284933
readaloud/api/smoke/heartbeat 1559284933
readaloud/api/camera-camp/heartbeat 1559284933
```

ps. mosquitto v.1.4.10 is getting a bit old.  it's from 2016!

**Comment by christianTF** [comment]
Thanks for the log. I'll give that a try on my DEV.
The subscription looks fine.

About Mosquitto version, it comes from the Debian repo.

**Comment by christianTF** [comment]
https://github.com/eclipse/mosquitto/issues/1249

**Comment by christianTF** [comment]
@emes Could you install the current master and give me feedback, if everything now/still works.
I've updated Net::MQTT::Simple to v1.23, that should fix problems with MQTT Standard V3.1.1

Thanks!

**Comment by emes** [comment]
Confirmed.  Thanks!
